src/display/font_loader.js

```diff
@@ -365,9 +365,41 @@ var FontFaceObject = (function FontFaceObjectClosure() {
 
     getPathGenerator: function FontLoader_getPathGenerator(objs, character) {
       if (!(character in this.compiledGlyphs)) {
-        var js = objs.get(this.loadedName + '_path_' + character);
-        /*jshint -W054 */
-        this.compiledGlyphs[character] = new Function('c', 'size', js);
+        var cmds = objs.get(this.loadedName + '_path_' + character);
+        // Try to compile cmds into JS for MAXIMUM SPEED
+        var js = '';
+        var cmd, args;
+        try {
+          for (var i in cmds) {
+            cmd = cmds[i][0];
+            args = cmds[i].slice(1);
+            js += 'c.' + cmd + '(' + args.join(',') + ');\n';
+          }
+          /* jshint -W054 */
+          this.compiledGlyphs[character] = new Function('c', 'size', js);
+        } catch(e) {
```

Comment:
Use lazy initialized property (e.g. `isEvalSupported`) at FontLoader that will use shadow() where you will test`new Function` capabilities and branch based on that, e.g. `var result; try { new Function(''); result = true; } catch (e) { result = false }; return shadow(...`.

See example at https://github.com/skalnik/pdf.js/blob/fix-font-csp-issues/src/display/font_loader.js#L54